### ISSUE TITLE
aot compile failed with "TypeError:arg is neither SymInt/int nor torch.Tensor, None" when run with TORCH_COMPILE_DEBUG=1

### ISSUE THREAD

**Issue by henrylhtsang** [pytorch/pytorch#135584]
### 🐛 Describe the bug

When used without `TORCH_COMPILE_DEBUG=1`, the following code is fine (it can run). When used with `TORCH_COMPILE_DEBUG=1`, it ran into the error "TypeError:arg is neither SymInt/int nor torch.Tensor, None
".

# repro:
```
import os

os.environ["TORCH_COMPILE_DEBUG"] = "1"

import torch


class Model(torch.nn.Module):
    def forward(self, x, ks0):
        return x.sum()


example_inputs = (
    torch.tensor([0, 3, 6], device="cuda", dtype=torch.int64),
    70,
)
_ = torch._export.aot_compile(
    Model(),
    example_inputs,
)
print("done")
```

# error 
```

/torch/_inductor/compile_fx.py:751, in fx_codegen_and_compile(gm, example_inputs, cudagraphs, static_input_idxs, is_backward, graph_id, cpp_wrapper, aot_mode, is_inference, user_visible_outputs, layout_opt, extern_node_serializer)
    740     return fd.getvalue()
    742 torch._logging.trace_structured(
    743     "artifact",
    744     metadata_fn=lambda: {
   (...)
    748     payload_fn=lambda: log_graph_runnable(),
    749 )
--> 751 V.debug.fx_graph(gm, example_inputs)
    752 # TODO: Should we actually dump this?  It should be redundant with the aot
    753 # structured logs...
    754 # trace_structured("inductor_input_graph", payload_fn=lambda: gm.print_readable(print_output=False))
    756 shape_env = shape_env_from_inputs(example_inputs)
/torch/_inductor/debug.py:474, in DebugFormatter.fx_graph(self, gm, inputs)
    468 def fx_graph(
    469     self,
    470     gm: torch.fx.GraphModule,
    471     inputs: List[torch.Tensor],
    472 ) -> None:
    473     with self.fopen("fx_graph_runnable.py") as fd:
--> 474         save_graph_repro(fd, gm, inputs, "inductor")
    476     with self.fopen("fx_graph_readable.py") as fd:
    477         fd.write(gm.print_readable(print_output=False))
/torch/_dynamo/repro/after_aot.py:279, in save_graph_repro(fd, gm, args, compiler_name, stable_output, save_dir, command, accuracy, tracing_mode, check_str)
    274     fd.write(
    275         "Repro is not generated due to existence of BackwardState in graph input"
    276     )
    277     return
    278 fd.write(
--> 279     generate_compiler_repro_string(
    280         gm,
    281         args,
    282         stable_output=stable_output,
    283         save_dir=save_dir,
    284     )
    285 )
    286 if accuracy is None:
    287     accuracy = "_accuracy" in compiler_name
/torch/_dynamo/repro/after_aot.py:249, in generate_compiler_repro_string(gm, args, stable_output, save_dir)
    247         writer.tensor(placeholder, arg)
    248     else:
--> 249         raise TypeError(f"arg is neither SymInt/int nor torch.Tensor, {arg}")
    251 model_str += "\n".join(writer.lines()) + "\n"
    253 model_str += "mod = Repro()\n"
TypeError: arg is neither SymInt/int nor torch.Tensor, None
```

### Versions

trunk

cc @ezyang @chauhang @penguinwu @avikchaudhuri @gmagogsfm @zhxchen17 @tugsbayasgalan @angelayi @suo @ydwu4

**Comment by cccntu** [comment]
I had the same issue. For me it happened when I set the env var `TORCH_TRACE`, and call `torch._export.aot_compile`
I think the issue comes from here: 
https://github.com/pytorch/pytorch/blob/663e7600652f042dffaa1061709f867d86b3a58e/torch/_inductor/__init__.py#L130-L134
In my case it converted a constant argument (False) into None.

Which eventually triggered the error here:
https://github.com/pytorch/pytorch/blob/663e7600652f042dffaa1061709f867d86b3a58e/torch/_dynamo/repro/after_aot.py#L241-L249


I was able to suppress the error by adding
```
        elif arg is None:
            continue
```
but it's probably not an ideal fix.

**Comment by ezyang** [comment]
This repros as of e4cdc31227de4ec20d8d30e03c2a396a19a9ec39

The problem is probably silly, just needs some looking...

**Comment by coconutruben** [comment]
This reproduces, thanks a bunch for the clean repro Henry!

seems to be added by https://github.com/pytorch/pytorch/commit/f063027d5424c6b90588ef0e84e9c21be4ce68ae

from the commit context, this does look like the intention is to fully skip it, but that's just conjecture

I'm trying to get more context, just started on this codebase, but from the outside it looks like for the graph compilation part, we want _constants_ to be filtered out, but the (debug) logger expects tensors or ints to log. Naively, let me ask the following

- is all we care about the logging here? if so, we can do what @cccntu suggested and just
  - skip the arg
  - write that there is another arg and it's None because it was filtered
  - create a new type, annotate it when filtering, and produce the right log lines e.g. _this is a constant bool_

- why is an integer as in the example above considered a constant in this scenario? we filter out everything that isn't a Tensor, is that intentional?

thanks for any pointers!

**Comment by angelayi** [comment]
Yup, we can skip constant inputs (like python primitives) since export [specializes](https://pytorch.org/docs/main/export.html#python-primitives) on these values and doesn't actually use the inputs passed into the graph.

I think we can just fix the logging to note that the arg is filtered out.

**Comment by henrylhtsang** [comment]
Yeah. My understanding is that constant ints as inputs won't be interpreted as symints, and they will just get hardcoded in the exported program.

If one wants to use symint, they should have it as a single element tensor, and use tensor.item().

**Comment by coconutruben** [comment]
this should've been fixed by #137032
